evo_researcher/benchmark/agents.py

```diff
@@ -195,6 +224,104 @@ def side_effect(*args: t.Any, **kwargs: t.Any) -> list[tuple[str, WebSearchResul
         with patch('evo_researcher.functions.research.search', side_effect=side_effect, autospec=True):
             return self.predict(market_question)
 
+    def research(
+        self,
+        goal: str,
+        use_summaries: bool,
+        model: str = "gpt-4-1106-preview",
+        initial_subqueries_limit: int = 20,
+        subqueries_limit: int = 4,
+        scrape_content_split_chunk_size: int = 800,
+        scrape_content_split_chunk_overlap: int = 225,
+        top_k_per_query: int = 8,
+        use_tavily_raw_content: bool = False,
+        openai_api_key: str | None = None,
+        tavily_api_key: str | None = None,
+    ) -> str:
```

Comment:
I think the idea for benchmark agents wasn't to hold the actual logic, only to implement the interface for benchmark. But @evangriffiths can correct me here.